mirascope/base/chunkers.py

```diff
@@ -0,0 +1,28 @@
+import uuid
+from typing import Callable, ClassVar
+
+from pydantic import BaseModel
+
+from mirascope.base.types import BaseChunkerParams, Document
+
+
+def base_chunker(text: str, params: BaseChunkerParams) -> list[Document]:
+    chunks: list[Document] = []
+    chunk_size = params.chunk_size or 0
+    chunk_overlap = params.chunk_overlap or 0
+    start: int = 0
+    while start < len(text):
+        end: int = min(start + chunk_size, len(text))
+        chunks.append(Document(text=text[start:end], id=str(uuid.uuid4())))
+        start += chunk_size - chunk_overlap
+    return chunks
+
+
+class BaseChunker(BaseModel):
+    chunker_params: ClassVar[BaseChunkerParams] = BaseChunkerParams(
+        chunk_size=1000, chunk_overlap=0
+    )
+    chunk_fn: Callable[[str, BaseChunkerParams], list[Document]] = base_chunker
```

Comment:
I feel like this should be a class variable? Also what do you think about `chunker` instead of `chunk_fn`? Or name it `base_chunk_fn` instead of `base_chunker`.

Reply:
oh yeah, it should be a classvar. is `chunker `clear enough?

Reply:
My only concern is overloading `chunker` and `Chunker`, so maybe `chunk_fn` is good and we should just use `base_chunk_fn`